bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -123,12 +123,12 @@ impl Connection {
         Ok(self)
     }
 
-    /// Reports the remaining nanoseconds before the connection may be safely closed.
+    /// Reports the remaining nanoseconds before the connection may be gracefully shutdown.
     ///
-    /// If [`poll_shutdown`] is called before this method reports "0", then an error will occur.
-    ///
-    /// This method is expected to succeed, but could fail if the underlying C call encounters errors.
-    /// If it fails, a graceful two-way shutdown of the connection will not be possible.
+    /// This method is expected to succeed, but could fail if the
+    /// [underlying C call](`s2n_connection_get_delay`) encounters errors.
+    /// Failure indicates that calls to [`Self::poll_shutdown`] will also fail and
+    /// that a graceful two-way shutdown of the connection will not be possible.
```

Comment:
nice, this makes more sense. does it also make sense to add/keep

`If the application calls [`poll_shutdown`] before this method reports "0", then an error will occur.`

Reply:
Nope, because that's not true anymore :) I [added a check](https://github.com/aws/s2n-tls/blob/main/bindings/rust/s2n-tls/src/connection.rs#L433-L435) to poll_shutdown a while back so that it returns Pending instead. The error was because that's how the underlying C shutdown method behaves, but now we don't call that method if it would fail due to blinding.